test/qbwc/integration/response_test.rb

```diff
@@ -104,57 +107,199 @@ def handle_response(resp, job, request, data)
 
   end
 
-  test "processes warning then error" do
-
-    # Add jobs
-    QBWC.add_job(:query_joe_customer,       true, COMPANY, HandleResponseWithDataWorker)
-    QBWC.add_job(:query_joe_customer_again, true, COMPANY, HandleResponseWithDataWorker)
+  def _single_request_helper(response, expected_status_code, expected_status_severity, expected_session_error)
+    error_string = "QBWC #{expected_status_severity.upcase}: #{expected_status_code} - #{expected_session_error}"
 
     # Simulate controller authenticate
     ticket_string = QBWC::ActiveRecord::Session.new(QBWC_USERNAME, COMPANY).ticket
     session = QBWC::Session.new(nil, COMPANY)
 
     # Simulate controller receive_response
-    session.response = QBWC_CUSTOMER_QUERY_RESPONSE_WARN
-    assert_equal '500',  session.status_code
-    assert_equal 'Warn', session.status_severity
-    assert_equal "QBWC WARN: 500 - #{QBWC_CUSTOMER_QUERY_STATUS_MESSAGE_WARN}", session.error
+    session.response = response
+    assert_equal 100,                      session.progress
+    assert_equal expected_status_code,     session.status_code
+    assert_equal expected_status_severity, session.status_severity
+    assert_equal error_string,             session.error
 
     # Simulate controller send_request
-    assert_not_nil session.next_request
-
-    # Simulate controller receive_response
-    session.response = QBWC_CUSTOMER_QUERY_RESPONSE_ERROR
-    assert_equal '3120',  session.status_code
-    assert_equal 'Error', session.status_severity
-    assert_equal "QBWC ERROR: 3120 - #{QBWC_CUSTOMER_QUERY_STATUS_MESSAGE_ERROR}", session.error
+    assert_nil session.next_request
   end
 
-  test "processes error then warning" do
-    QBWC.on_error = :continue
+  def _multi_request_helper(response1, expected_status_code1, expected_status_severity1, expected_session_error1, response2, expected_status_code2, expected_status_severity2, expected_session_error2, expected_progress1 = 50)
```

Comment:
Can you change this method to accept any number of requests? Maybe `_multi_request_helper(*hashes)`, where each individual hash is `{:response, :code, :severity, :error}`. This would make it more useful and the calls easier to read.